packages/debug/addon/index.js

```diff
@@ -38,15 +38,72 @@ export default DataAdapter.extend({
     ];
   },
 
+  _nameToClass(type) {
+    return get(this, 'store').modelFor(type);
+  },
+
   /**
-    Detect whether a class is a Model
-    @private
-    @method detect
-    @param {Model} typeClass
-    @return {Boolean} Whether the typeClass is a Model class or not
+    Fetch the model types and observe them for changes.
+    Maintains the list of model types without needing the Model package for detection.
+    @public
+    @method watchModelTypes
+    @param {Function} typesAdded Callback to call to add types.
+    Takes an array of objects containing wrapped types (returned from `wrapModelType`).
+    @param {Function} typesUpdated Callback to call when a type has changed.
+    Takes an array of objects containing wrapped types.
+    @return {Function} Method to call to remove all observers
   */
-  detect(typeClass) {
-    return typeClass !== Model && Model.detect(typeClass);
+  watchModelTypes(typesAdded, typesUpdated) {
+    const store = get(this, 'store');
+    const __createRecordData = store._createRecordData;
+    const _releaseMethods = [];
+    const discoveredTypes = StoreTypesMap.get(store);
+
+    // Add any models that were added during initialization of the app, before the inspector was opened
+    discoveredTypes.forEach((_, type) => {
+      this.watchTypeIfUnseen(store, discoveredTypes, type, typesAdded, typesUpdated, _releaseMethods);
+    });
+
+    // Overwrite _createRecordData so newly added models will get added to the list
+    store._createRecordData = function(identifier) {
```

Comment:
this should be an arrow function to preserve the `this` context of the data adapter
```suggestion
    store._createRecordData = (identifier) => {
```